torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -895,29 +901,50 @@ struct CudaGraphFuser {
       }
       // TODO: output(1) & output(2) should also be marked
       if (n->kind() == aten::native_layer_norm) {
+        TORCH_INTERNAL_ASSERT(
+            shape_of.count(n->input(0)) > 0,
+            "buildShapeExpressions failed at accessing input shapes");
         shape_of.emplace(n->output(0), shape_of.at(n->input(0)));
         continue;
       }
       // TODO: output(1) & output(2) should also be marked
       if (n->kind() == aten::native_layer_norm_backward) {
+        TORCH_INTERNAL_ASSERT(
+            shape_of.count(n->input(0)) > 0,
+            "buildShapeExpressions failed at accessing input shapes");
         shape_of.emplace(n->output(0), shape_of.at(n->input(0)));
```

Comment:
Shouldn't you check all of the locations you're accessing? Is it guaranteed if `n->input(0)` is present in `shape_of` that `n->input(5)` and `n->input(6)` will be too?

Reply:
Note to myself, same beneath for `native_batch_norm_backward`. Do not assert on those, but just leave them empty instead (since grads could be omitted)